enctests/testframework/generatetests.py

```diff
@@ -0,0 +1,144 @@
+# This is for generating tests from the markdown code.
+import os
+import argparse
+import yaml
+import re
+from pathlib import Path
+
+try:
+    from yaml import CSafeLoader as SafeLoader
+    from yaml import CSafeDumper as SafeDumper
+
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--root',
+        action='store',
+        default='..',
+        help='Where to look for *.md files to generate the config file.'
+    )
+
+    parser.add_argument(
+        '--output-config',
+        action='store',
+        dest='output_config_file',
+        default="test_configs/documentation_tests.yml",
+        help='Specify output file to write test wrappers to.'
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+def main():
+    args = parse_args()
+    rootdir = Path(args.root).absolute()
+
+    markdownfiles = []
+    for root, dirs, files in os.walk(rootdir):
+        for name in files:
+            if name.endswith(".md"):
+                markdownfiles.append(os.path.join(root, name))
+
+    allinfo = []
+
+    for markdownfile in markdownfiles:
+        with open(markdownfile, "r") as f:
+            incomment = False
+            incommand = False
+            command = ""
+            info = ""
+            lastcommandstart = -1
+            for linenumber, line in enumerate(f):
+                if "<!---" in line:
+                    incomment = True
+                    continue
+                if "-->" in line:
+                    incomment = False
+                    continue
+                if "```" in line:
+                    if incommand:
+                        incommand = False
+                        if "ffmpeg" in command and info != "":
+                            try:
+                                infostruct = yaml.load(info, SafeLoader)
+                                command = command.replace("\\\n", "").replace("\n", "")
+                                allinfo.append({'config': infostruct, 'command': command, 'file': markdownfile, 'line': lastcommandstart})
+                            except:
+                                pass
+                        command = ""
+                        info = ""
+                    else:
+                        incommand = True
+                        lastcommandstart = linenumber
+                    continue
+                if incomment:
+                    info = info + line
+                if incommand:
+                    command = command + line
+
+    # Now we process the scanned data to create the config file.
+    tests = []
+
+    for testcount, test in enumerate(allinfo):
```

Comment:
Should we make this a function like the suggested markdown parsing function?

That would leave `main()` quite clean